packages/console/src/components/CreateTenantModal/SelectTenantPlanModal/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { type TenantInfo } from '@logto/schemas/models';
+import { toast } from 'react-hot-toast';
+import { Trans, useTranslation } from 'react-i18next';
+import Modal from 'react-modal';
+
+import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { ReservedPlanId } from '@/consts/subscriptions';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import ModalLayout from '@/ds-components/ModalLayout';
+import TextLink from '@/ds-components/TextLink';
+import useSubscriptionPlans from '@/hooks/use-subscription-plans';
+import * as modalStyles from '@/scss/modal.module.scss';
+import { type SubscriptionPlan } from '@/types/subscriptions';
+
+import { type CreateTenantData } from '../type';
+
+import PlanCardItem from './PlanCardItem';
+import * as styles from './index.module.scss';
+
+type Props = {
+  createTenantData?: CreateTenantData;
+  onClose: (tenant?: TenantInfo) => void;
+};
+
+function SelectTenantPlanModal({ createTenantData, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { data: subscriptionPlans } = useSubscriptionPlans();
+  const cloudApi = useCloudApi();
+  if (!subscriptionPlans || !createTenantData) {
+    return null;
+  }
+
+  const handleSelectPlan = async (plan: SubscriptionPlan) => {
+    const { id: planId } = plan;
+    if (planId === ReservedPlanId.free) {
+      try {
+        const { name, tag } = createTenantData;
+        const newTenant = await cloudApi.post('/api/tenants', { body: { name, tag } });
+
+        onClose(newTenant);
+      } catch (error: unknown) {
+        toast.error(error instanceof Error ? error.message : String(error));
+      }
+    }
+    // Todo @xiaoyijun implement checkout
+  };
+
+  return (
+    <Modal
+      shouldCloseOnEsc
+      isOpen={Boolean(createTenantData)}
+      className={modalStyles.content}
+      overlayClassName={modalStyles.overlay}
+      onRequestClose={() => {
+        onClose();
+      }}
```

Comment:
nit: `onRequestClose={onClose}`

Reply:
Their types are not the same.
<img width="826" alt="image" src="https://github.com/logto-io/logto/assets/10806653/00e69f20-d266-41a5-b684-7be627de74c0">